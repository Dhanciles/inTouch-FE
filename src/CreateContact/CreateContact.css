.new-contact-header {
  margin-top: 2em;
  font-family: 'Oswald', sans-serif;
  font-size: 1.5em;
  font-weight: 300;
  color: #191C69;
  text-align: center;
  margin-right: 50px;
  margin-bottom: 1em;
}

.article {
  margin-top: 1em;
  margin: auto 0;
  width: 90%;
  height: 370px;
  border-radius: 20px;
  border: 2px solid #191C69;
}

form {
  margin-right: 65px;
}

input {
  line-height: 1rem; 
  background: transparent;
  text-align: left;
  margin-left: 10px;
  font-family: 'Oswald', sans-serif;
  font-weight: 100; 
  font-size: 1.5rem;
  color: #191C69; 
  border: none;
  border-bottom: 2px solid #191C69;
}

.name {
  height: 50px;
  width: 20px;
  margin-left: 20px;
  margin-bottom: 4em;
}

.priority {
  margin-top: 1em;
  margin-left: 30px;
  margin-bottom: 0.5;
  height: 150;
  width: 200px;
}

.priority-selection {
  height: 30px;
  width: 200px
}

.frequency {
  display: flex;
  margin-top: 0.8em;
  margin-left: 20px;
  margin-bottom: 0;
}

.frequency-input {
  margin-top: 0.2em;
  height: 40px;
  width: 50px;
  text-align: center;
}

p {
  margin-left: 10px;
  font-family: 'Oswald', sans-serif;
  font-size: 1.2rem;
  font-weight: 300;
  color: #191C69;
}

::placeholder {
  color: #191C69; 
  font-size: 1.2rem;
  margin-left: 10px;
}

label {
  margin-top: 1em;
  margin-left: 30px;
  margin-bottom: 3em;
  font-family: 'Oswald', sans-serif;
  font-size: 1.2rem;
  font-weight: 300;
  color: #191C69;
}

.notes {
  width: 445px;
  height: 100px;
  margin-left: 10%;
  border-radius: 20px;
  background-color: #ABAECF;
}

.save-contact {
  text-align: center;
  margin-top: 1em;
  margin-right: 50px;
}

button {
  width: 10em;
  height: 2.2em;
  border-radius: 40px;
  background-color: #5B609F;
  color: #FFFFFF;
  font-size: 1em;
}

button:hover {
  background-color: #191C69;
}