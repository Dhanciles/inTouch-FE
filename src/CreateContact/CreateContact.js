import React, { Component } from 'react'; 
import './CreateContact.css'
import { addContact } from '../Thunks/addContact.js'
import { isLoading, hasErrored } from '../actions/index.js'
import { connect } from 'react-redux'


export class CreateContact extends Component {
  constructor() {
    super()
    this.state = {
     name: '',
     frequency: '',
     priority: '',
     notes: '', 
     disabled: true
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({
      [name]: value
    }, () => this.enableButton())
  }

  enableButton = () => {
    const { name, frequency, priority } = this.state
    if (name !== '' && frequency !== '' && priority !== '') {
      this.setState({
        disabled: false
      })
    }
  }

  handleSubmit = (e) => {
    e.preventDefault()
    const { name, frequency, priority } = this.state
    this.props.addContact(name, frequency, priority, this.props.token)
  }
  
  render() {
    const { name, frequency, priority, notes } = this.state
    return (
      <form>
        <div className="new-contact-header">New Contact</div>
        <div className="article">
          <div className="name">
            <input className="new-contact-name" name="contactName" value={name} type="text" required placeholder="Name"/>
          </div>
          <div className="priority">
            <select className="priority-selection" value={priority} required> 
              <option value="prio-1">1 - least important</option>
              <option value="prio-2">2</option>
              <option value="prio-3">3</option>
              <option value="prio-4">4</option>
              <option value="prio-5">5 - most important</option>
            </select>
          </div>
          <div className="frequency"> 
            <p>Contact every</p>
            <input className="frequency-input" name="contactFrequency" value={priority} type="text" required required placeholder="7"/> 
            <p>days</p>
          </div>
          <label className="notes-label">Notes:</label>
          <div className="notes">
            <text-area value={notes} rows="5" className="note-field" placeholder="Take your notes here."/>  
          </div>          
        </div>
        <div className="save-contact">
          <button className='save-contact-btn'>Save Contact</button>
        </div>
      </form>
    )
  }
}

export const mapStateToProps = (state) => ({
  token: state.user.token, 
  error: state.errored, 
  loading: state.loading
})

export const mapDispatchToProps = (dispatch) => ({
  addContact: (name, frequency, priority, token) => dispatch(addContact(name, frequency, priority, token))
})

export default connect(mapStateToProps, mapDispatchToProps)(CreateContact)