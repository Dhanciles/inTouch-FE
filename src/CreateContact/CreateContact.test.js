import React from 'react'; 
import { shallow } from 'enzyme'; 
import CreateContact from './CreateContact.js'; 

describe('CreateContact', () => {
  let mockName; 
  let mockContactType; 
  let mockContactInformation; 
  let mockFrequency; 
  let mockPriority; 
  let mockNotes; 

  beforeEach(() => {
    mockName = 'Rajaa'; 
    mockContactType = 'Email'; 
    mockContactInformation = 'rajaa@email'; 
    mockFrequency = 'weekly'; 
    mockPriority = 3; 
    mockNotes = ''; 
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<CreateContact />)

    expect(wrapper).toMatchSnapshot()
  });
});