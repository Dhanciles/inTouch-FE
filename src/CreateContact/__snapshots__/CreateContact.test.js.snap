// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateContact should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateContact />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="new-contact-header"
        >
          New Contact
        </div>,
        <div
          className="article"
        >
          <div
            className="name"
          >
            <input
              className="new-contact-name"
              name="contactName"
              placeholder="Name"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="priority"
          >
            <select
              className="priority-selection"
              required={true}
              value=""
            >
              <option
                value="prio-1"
              >
                1 - least important
              </option>
              <option
                value="prio-2"
              >
                2
              </option>
              <option
                value="prio-3"
              >
                3
              </option>
              <option
                value="prio-4"
              >
                4
              </option>
              <option
                value="prio-5"
              >
                5 - most important
              </option>
            </select>
          </div>
          <div
            className="frequency"
          >
            <p>
              Contact every
            </p>
            <input
              className="frequency-input"
              name="contactFrequency"
              placeholder="7"
              required={true}
              type="text"
              value=""
            />
            <p>
              days
            </p>
          </div>
          <label
            className="notes-label"
          >
            Notes:
          </label>
          <div
            className="notes"
          >
            <text-area
              className="note-field"
              placeholder="Take your notes here."
              rows="5"
              value=""
            />
          </div>
        </div>,
        <div
          className="save-contact"
        >
          <button
            className="save-contact-btn"
          >
            Save Contact
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "New Contact",
          "className": "new-contact-header",
        },
        "ref": null,
        "rendered": "New Contact",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="name"
            >
              <input
                className="new-contact-name"
                name="contactName"
                placeholder="Name"
                required={true}
                type="text"
                value=""
              />
            </div>,
            <div
              className="priority"
            >
              <select
                className="priority-selection"
                required={true}
                value=""
              >
                <option
                  value="prio-1"
                >
                  1 - least important
                </option>
                <option
                  value="prio-2"
                >
                  2
                </option>
                <option
                  value="prio-3"
                >
                  3
                </option>
                <option
                  value="prio-4"
                >
                  4
                </option>
                <option
                  value="prio-5"
                >
                  5 - most important
                </option>
              </select>
            </div>,
            <div
              className="frequency"
            >
              <p>
                Contact every
              </p>
              <input
                className="frequency-input"
                name="contactFrequency"
                placeholder="7"
                required={true}
                type="text"
                value=""
              />
              <p>
                days
              </p>
            </div>,
            <label
              className="notes-label"
            >
              Notes:
            </label>,
            <div
              className="notes"
            >
              <text-area
                className="note-field"
                placeholder="Take your notes here."
                rows="5"
                value=""
              />
            </div>,
          ],
          "className": "article",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="new-contact-name"
                name="contactName"
                placeholder="Name"
                required={true}
                type="text"
                value=""
              />,
              "className": "name",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "new-contact-name",
                "name": "contactName",
                "placeholder": "Name",
                "required": true,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <select
                className="priority-selection"
                required={true}
                value=""
              >
                <option
                  value="prio-1"
                >
                  1 - least important
                </option>
                <option
                  value="prio-2"
                >
                  2
                </option>
                <option
                  value="prio-3"
                >
                  3
                </option>
                <option
                  value="prio-4"
                >
                  4
                </option>
                <option
                  value="prio-5"
                >
                  5 - most important
                </option>
              </select>,
              "className": "priority",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="prio-1"
                  >
                    1 - least important
                  </option>,
                  <option
                    value="prio-2"
                  >
                    2
                  </option>,
                  <option
                    value="prio-3"
                  >
                    3
                  </option>,
                  <option
                    value="prio-4"
                  >
                    4
                  </option>,
                  <option
                    value="prio-5"
                  >
                    5 - most important
                  </option>,
                ],
                "className": "priority-selection",
                "required": true,
                "value": "",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1 - least important",
                    "value": "prio-1",
                  },
                  "ref": null,
                  "rendered": "1 - least important",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2",
                    "value": "prio-2",
                  },
                  "ref": null,
                  "rendered": "2",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "3",
                    "value": "prio-3",
                  },
                  "ref": null,
                  "rendered": "3",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "4",
                    "value": "prio-4",
                  },
                  "ref": null,
                  "rendered": "4",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "5 - most important",
                    "value": "prio-5",
                  },
                  "ref": null,
                  "rendered": "5 - most important",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Contact every
                </p>,
                <input
                  className="frequency-input"
                  name="contactFrequency"
                  placeholder="7"
                  required={true}
                  type="text"
                  value=""
                />,
                <p>
                  days
                </p>,
              ],
              "className": "frequency",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Contact every",
                },
                "ref": null,
                "rendered": "Contact every",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "frequency-input",
                  "name": "contactFrequency",
                  "placeholder": "7",
                  "required": true,
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "days",
                },
                "ref": null,
                "rendered": "days",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Notes:",
              "className": "notes-label",
            },
            "ref": null,
            "rendered": "Notes:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <text-area
                className="note-field"
                placeholder="Take your notes here."
                rows="5"
                value=""
              />,
              "className": "notes",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "note-field",
                "placeholder": "Take your notes here.",
                "rows": "5",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "text-area",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="save-contact-btn"
          >
            Save Contact
          </button>,
          "className": "save-contact",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save Contact",
            "className": "save-contact-btn",
          },
          "ref": null,
          "rendered": "Save Contact",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="new-contact-header"
          >
            New Contact
          </div>,
          <div
            className="article"
          >
            <div
              className="name"
            >
              <input
                className="new-contact-name"
                name="contactName"
                placeholder="Name"
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="priority"
            >
              <select
                className="priority-selection"
                required={true}
                value=""
              >
                <option
                  value="prio-1"
                >
                  1 - least important
                </option>
                <option
                  value="prio-2"
                >
                  2
                </option>
                <option
                  value="prio-3"
                >
                  3
                </option>
                <option
                  value="prio-4"
                >
                  4
                </option>
                <option
                  value="prio-5"
                >
                  5 - most important
                </option>
              </select>
            </div>
            <div
              className="frequency"
            >
              <p>
                Contact every
              </p>
              <input
                className="frequency-input"
                name="contactFrequency"
                placeholder="7"
                required={true}
                type="text"
                value=""
              />
              <p>
                days
              </p>
            </div>
            <label
              className="notes-label"
            >
              Notes:
            </label>
            <div
              className="notes"
            >
              <text-area
                className="note-field"
                placeholder="Take your notes here."
                rows="5"
                value=""
              />
            </div>
          </div>,
          <div
            className="save-contact"
          >
            <button
              className="save-contact-btn"
            >
              Save Contact
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New Contact",
            "className": "new-contact-header",
          },
          "ref": null,
          "rendered": "New Contact",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="name"
              >
                <input
                  className="new-contact-name"
                  name="contactName"
                  placeholder="Name"
                  required={true}
                  type="text"
                  value=""
                />
              </div>,
              <div
                className="priority"
              >
                <select
                  className="priority-selection"
                  required={true}
                  value=""
                >
                  <option
                    value="prio-1"
                  >
                    1 - least important
                  </option>
                  <option
                    value="prio-2"
                  >
                    2
                  </option>
                  <option
                    value="prio-3"
                  >
                    3
                  </option>
                  <option
                    value="prio-4"
                  >
                    4
                  </option>
                  <option
                    value="prio-5"
                  >
                    5 - most important
                  </option>
                </select>
              </div>,
              <div
                className="frequency"
              >
                <p>
                  Contact every
                </p>
                <input
                  className="frequency-input"
                  name="contactFrequency"
                  placeholder="7"
                  required={true}
                  type="text"
                  value=""
                />
                <p>
                  days
                </p>
              </div>,
              <label
                className="notes-label"
              >
                Notes:
              </label>,
              <div
                className="notes"
              >
                <text-area
                  className="note-field"
                  placeholder="Take your notes here."
                  rows="5"
                  value=""
                />
              </div>,
            ],
            "className": "article",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  className="new-contact-name"
                  name="contactName"
                  placeholder="Name"
                  required={true}
                  type="text"
                  value=""
                />,
                "className": "name",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "new-contact-name",
                  "name": "contactName",
                  "placeholder": "Name",
                  "required": true,
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <select
                  className="priority-selection"
                  required={true}
                  value=""
                >
                  <option
                    value="prio-1"
                  >
                    1 - least important
                  </option>
                  <option
                    value="prio-2"
                  >
                    2
                  </option>
                  <option
                    value="prio-3"
                  >
                    3
                  </option>
                  <option
                    value="prio-4"
                  >
                    4
                  </option>
                  <option
                    value="prio-5"
                  >
                    5 - most important
                  </option>
                </select>,
                "className": "priority",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value="prio-1"
                    >
                      1 - least important
                    </option>,
                    <option
                      value="prio-2"
                    >
                      2
                    </option>,
                    <option
                      value="prio-3"
                    >
                      3
                    </option>,
                    <option
                      value="prio-4"
                    >
                      4
                    </option>,
                    <option
                      value="prio-5"
                    >
                      5 - most important
                    </option>,
                  ],
                  "className": "priority-selection",
                  "required": true,
                  "value": "",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "1 - least important",
                      "value": "prio-1",
                    },
                    "ref": null,
                    "rendered": "1 - least important",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "2",
                      "value": "prio-2",
                    },
                    "ref": null,
                    "rendered": "2",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "3",
                      "value": "prio-3",
                    },
                    "ref": null,
                    "rendered": "3",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "4",
                      "value": "prio-4",
                    },
                    "ref": null,
                    "rendered": "4",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "5 - most important",
                      "value": "prio-5",
                    },
                    "ref": null,
                    "rendered": "5 - most important",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Contact every
                  </p>,
                  <input
                    className="frequency-input"
                    name="contactFrequency"
                    placeholder="7"
                    required={true}
                    type="text"
                    value=""
                  />,
                  <p>
                    days
                  </p>,
                ],
                "className": "frequency",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Contact every",
                  },
                  "ref": null,
                  "rendered": "Contact every",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "frequency-input",
                    "name": "contactFrequency",
                    "placeholder": "7",
                    "required": true,
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "days",
                  },
                  "ref": null,
                  "rendered": "days",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Notes:",
                "className": "notes-label",
              },
              "ref": null,
              "rendered": "Notes:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <text-area
                  className="note-field"
                  placeholder="Take your notes here."
                  rows="5"
                  value=""
                />,
                "className": "notes",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "note-field",
                  "placeholder": "Take your notes here.",
                  "rows": "5",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "text-area",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="save-contact-btn"
            >
              Save Contact
            </button>,
            "className": "save-contact",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save Contact",
              "className": "save-contact-btn",
            },
            "ref": null,
            "rendered": "Save Contact",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
